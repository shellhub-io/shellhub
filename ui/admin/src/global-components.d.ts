// src/global-components.d.ts
import SnackbarComponent from "./components/Snackbar/Snackbar.vue";

declare module "@vue/runtime-core" {
  export interface GlobalComponents {
    SnackbarComponent: typeof SnackbarComponent;
    RouterLink: typeof import("vue-router")["RouterLink"];
    RouterView: typeof import("vue-router")["RouterView"];
  }
}
