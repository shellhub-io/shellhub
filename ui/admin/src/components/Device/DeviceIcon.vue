<template>
  <i :class="iconName" style="font-size: 20px" />
</template>

<script lang="ts">
import { computed, defineComponent } from "vue";

export default defineComponent({
  props: {
    icon: {
      type: String,
      required: true,
    },
  },
  setup(props) {
    const deviceIcon = {
      alpine: "fl-alpine",
      arch: "fl-archlinux",
      centos: "fl-centos",
      coreos: "fl-coreos",
      debian: "fl-debian",
      devuan: "fl-devuan",
      elementary: "fl-elementary",
      fedora: "fl-fedora",
      freebsd: "fl-freebsd",
      gentoo: "fl-gentoo",
      linuxmint: "fl-linuxmint",
      mageia: "fl-mageia",
      manjaro: "fl-manjaro",
      mandriva: "fl-mandriva",
      nixos: "fl-nixos",
      opensuse: "fl-opensuse",
      rhel: "fl-redhat",
      sabayon: "fl-sabayon",
      slackware: "fl-slackware",
      ubuntu: "fl-ubuntu",
      raspbian: "fl-raspberry-pi",
      "ubuntu-core": "fl-ubuntu",
      void: "fl-void",
    };

    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    const iconName = computed(() => deviceIcon[props.icon]);

    return {
      iconName,
    };
  },
});
</script>
