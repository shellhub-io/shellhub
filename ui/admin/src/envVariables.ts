/* eslint-disable import/prefer-default-export */
export const envVariables = {
  announcementsEnable: process.env.SHELLHUB_ANNOUNCEMENTS === "true",
  tinyMceKey: process.env.TINYMCE_API_KEY,
};
