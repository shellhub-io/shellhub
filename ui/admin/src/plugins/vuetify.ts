// Styles
// import "@mdi/font/css/materialdesignicons.css";
import "vuetify/styles";

// Vuetify
import { createVuetify } from "vuetify";

export default createVuetify({
  theme: {
    defaultTheme: "dark",
  },
});
