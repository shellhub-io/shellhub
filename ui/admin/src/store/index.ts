import { InjectionKey } from "vue";
import { createStore, Store, useStore as vuexUseStore } from "vuex";
import { stats, StatsState } from "./modules/stats";

import { UsersState, users } from "./modules/users";
import { AuthState, auth } from "./modules/auth";
import { DevicesState, devices } from "./modules/devices";
import { firewallRules, FirewallRulesState } from "./modules/firewall_rules";
import { namespaces, NamespacesState } from "./modules/namespaces";
import { layout, LayoutState } from "./modules/layout";
import { license, LicenseState } from "./modules/license";
import { publicKeys, PublicKeysState } from "./modules/public_keys";
import { sessions, SessionsState } from "./modules/sessions";
import { snackbar, SnackbarState } from "./modules/snackbar";
import { spinner, SpinnerState } from "./modules/spinner";
import { announcement, AnnouncementState } from "./modules/announcement";
import { instance, InstanceState } from "./modules/instance";

export interface State {
  user: UsersState;
  auth: AuthState;
  devices: DevicesState;
  firewallRules: FirewallRulesState;
  layout: LayoutState;
  license: LicenseState;
  namespaces: NamespacesState;
  publicKeys: PublicKeysState;
  sessions: SessionsState;
  snackbar: SnackbarState;
  spinner: SpinnerState;
  stats: StatsState;
  announcement: AnnouncementState;
  instance: InstanceState;
}

// eslint-disable-next-line symbol-description
export const key: InjectionKey<Store<State>> = Symbol();

export const store = createStore<State>({
  modules: {
    users,
    auth,
    devices,
    firewallRules,
    layout,
    license,
    namespaces,
    publicKeys,
    sessions,
    snackbar,
    spinner,
    stats,
    announcement,
    instance,
  },
});

export function useStore(): Store<State> {
  return vuexUseStore(key);
}
