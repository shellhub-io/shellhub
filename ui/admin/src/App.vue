<template>
  <v-app>
    <component
      :is="layout"
      :data-test="`${layout}-component`" />
  </v-app>
</template>

<script setup lang="ts">
import { computed } from "vue";
import SimpleLayout from "./layouts/SimpleLayout.vue";
import AppLayout from "./layouts/AppLayout.vue";
import { useStore } from "./store";

const components = {
  SimpleLayout,
  AppLayout,
};

const store = useStore();

const layout = computed(() => components[store.getters["layout/getLayout"]]);
</script>
