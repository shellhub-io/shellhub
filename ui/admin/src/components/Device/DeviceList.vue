<template>
  <div>
    <DataTable
      :headers="headers"
      :items="devices"
      :itemsPerPage="itemsPerPage"
      :nextPage="next"
      :previousPage="prev"
      :loading="loading"
      :totalCount="numberDevices"
      :actualPage="page"
      :comboboxOptions="[10, 20, 50, 100]"
      @changeItemsPerPage="changeItemsPerPage"
      @clickNextPage="next"
      @clickPreviousPage="prev"
      @clickSortableIcon="sortByItem"
      data-test="devices-list"
    >
      <template v-slot:rows>
        <tr v-for="(item, i) in devices" :key="i">
          <td>
            <v-icon v-if="item.online" color="success" data-test="sucess-icon">
              mdi-check-circle
            </v-icon>
            <v-icon v-else color="#E53935" data-test="error-icon"> mdi-close-circle </v-icon>
          </td>
          <td>{{ item.name }}</td>
          <td class="d-flex align-center">
            <DeviceIcon :icon="item.info.id" class="mr-2" />
            {{ item.info.prettyName }}
          </td>
          <td>
            <span
              @click="goToNamespace(item.tenant_id)"
              @keypress.enter="goToNamespace(item.tenant_id)"
              tabindex="0"
              class="hover"
            >
              {{ item.namespace }}
            </span>
          </td>
          <td>
            <div v-if="item.tags[0]">
              <v-tooltip
                v-for="(tag, index) in item.tags"
                :key="index"
                bottom
                :disabled="!showTag(tag)"
              >
                <template #activator="{ props }">
                  <v-chip size="small" v-bind="props" v-on="props">
                    {{ displayOnlyTenCharacters(tag) }}
                  </v-chip>
                </template>

                <span v-if="showTag(tag)">
                  {{ tag }}
                </span>
              </v-tooltip>
            </div>
          </td>
          <td>
            {{ formatDate(item.last_seen) }}
          </td>
          <td>
            <v-chip size="small">
              {{ item.status }}
            </v-chip>
          </td>
          <td>
            <v-tooltip bottom anchor="bottom">
              <template v-slot:activator="{ props }">
                <v-icon
                  tag="a"
                  dark
                  v-bind="props"
                  @click="redirectToDevice(item.uid)"
                  @keypress.enter="redirectToDevice(item.uid)"
                  tabindex="0"
                >mdi-information
                </v-icon>
              </template>
              <span>Info</span>
            </v-tooltip>
          </td>
        </tr>
      </template>
    </DataTable>
  </div>
</template>

<script lang="ts">
import { defineComponent, ref, onMounted, watch, computed } from "vue";
import { useRouter } from "vue-router";
import { useStore } from "../../store";
import DataTable from "../DataTable.vue";
import DeviceIcon from "./DeviceIcon.vue";
import { formatDate } from "../../hooks/formateDate";
import displayOnlyTenCharacters from "../../hooks/string";
import showTag from "../../hooks/tag";
import { INotificationsError } from "../../interfaces/INotifications";

export default defineComponent({
  setup() {
    const store = useStore();
    const router = useRouter();
    const loading = ref(false);
    const filter = ref("");
    const itemsPerPage = ref(10);
    const page = ref(1);

    const devices = computed(() => store.getters["devices/list"]);
    const numberDevices = computed(() => store.getters["devices/numberDevices"]);

    onMounted(async () => {
      try {
        loading.value = true;
        await store.dispatch("devices/fetch", {
          perPage: itemsPerPage.value,
          page: 1,
          filter: "",
          sortStatusField: "",
          sortStatusString: "",
        });
      } catch {
        store.dispatch("snackbar/showSnackbarErrorAction", INotificationsError.deviceList);
      } finally {
        loading.value = false;
      }
    });

    const getDevices = async (perPagaeValue: number, pageValue: number) => {
      try {
        loading.value = true;

        const hasDevices = await store.dispatch("devices/fetch", {
          perPage: perPagaeValue,
          page: pageValue,
          filter: filter.value,
          sortStatusField: store.getters["devices/sortStatusField"],
          sortStatusString: store.getters["devices/sortStatusString"],
        });

        if (!hasDevices) {
          page.value--;
        }

        loading.value = false;
      } catch (error) {
        store.dispatch("snackbar/showSnackbarErrorAction", INotificationsError.deviceList);
      }
    };

    const sortByItem = async (field: string) => {
      let sortStatusString = store.getters["devices/sortStatusString"];
      if (sortStatusString === "") {
        sortStatusString = "asc";
      } else if (sortStatusString === "asc") {
        sortStatusString = "desc";
      } else {
        sortStatusString = "";
      }
      await store.dispatch("devices/setSortStatus", {
        sortStatusField: field,
        sortStatusString,
      });
      await getDevices(itemsPerPage.value, page.value);
    };

    const next = async () => {
      await getDevices(itemsPerPage.value, ++page.value);
    };

    const prev = async () => {
      try {
        if (page.value > 1) await getDevices(itemsPerPage.value, --page.value);
      } catch (error) {
        store.dispatch("snackbar/setSnackbarErrorDefault");
      }
    };

    const changeItemsPerPage = async (newItemsPerPage: number) => {
      itemsPerPage.value = newItemsPerPage;
    };

    watch(itemsPerPage, async () => {
      await getDevices(itemsPerPage.value, page.value);
    });

    const goToNamespace = (namespace: string) => {
      router.push({ name: "namespaceDetails", params: { id: namespace } });
    };

    const redirectToDevice = (deviceId: string) => {
      router.push({ name: "deviceDetails", params: { id: deviceId } });
    };

    return {
      headers: [
        {
          text: "Online",
          value: "online",
          sortable: true,
        },
        {
          text: "Hostname",
          value: "name",
          sortable: true,
        },
        {
          text: "Info",
          value: "info",
          sortable: true,
        },
        {
          text: "Namespace",
          value: "namespace",
          sortable: true,
        },
        {
          text: "Tags",
          value: "tags",
        },
        {
          text: "Last Seen",
          value: "last_seen",
          sortable: true,
          align: "center",
        },
        {
          text: "Status",
          value: "status",
          sortable: true,
        },
        {
          text: "Actions",
          value: "actions",
        },
      ],
      itemsPerPage,
      page,
      loading,
      devices,
      numberDevices,
      next,
      prev,
      sortByItem,
      showTag,
      displayOnlyTenCharacters,
      formatDate,
      goToNamespace,
      redirectToDevice,
      changeItemsPerPage,
    };
  },
  components: { DataTable, DeviceIcon },
});
</script>

<style scoped>
.hover:hover,
.hover:focus {
  cursor: pointer;
  text-decoration: underline;
}
</style>
