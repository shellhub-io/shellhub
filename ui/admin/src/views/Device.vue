<template>
  <div class="d-flex flex-column justify-space-between align-center flex-sm-row">
    <h1>Devices</h1>
    <v-spacer />
    <div class="w-50">
      <v-text-field
        label="Search by hostname"
        variant="underlined"
        color="primary"
        single-line
        hide-details
        v-model.trim="filter"
        v-on:keyup="searchDevices"
        append-inner-icon="mdi-magnify"
        density="comfortable"
      />
    </div>
    <v-spacer />
  </div>
  <v-card class="mt-2">
    <DeviceList />
  </v-card>
</template>

<script setup lang="ts">
import { ref } from "vue";
import useSnackbarStore from "@admin/store/modules/snackbar";
import useDevicesStore from "@admin/store/modules/devices";
import DeviceList from "../components/Device/DeviceList.vue";

const snackbarStore = useSnackbarStore();
const devicesStore = useDevicesStore();

const filter = ref("");

const searchDevices = () => {
  let encodedFilter = "";

  if (filter.value) {
    const filterToEncodeBase64 = [
      {
        type: "property",
        params: { name: "name", operator: "contains", value: filter.value },
      },
    ];
    encodedFilter = btoa(JSON.stringify(filterToEncodeBase64));
  }

  try {
    devicesStore.search({
      perPage: devicesStore.getPerPage,
      page: devicesStore.getPage,
      filter: encodedFilter,
    });
  } catch {
    snackbarStore.showSnackbarErrorDefault();
  }
};

defineExpose({ filter });
</script>
