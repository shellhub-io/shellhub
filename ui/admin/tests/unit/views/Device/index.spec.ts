import { createStore } from "vuex";
import { createVuetify } from "vuetify";
import { mount, VueWrapper } from "@vue/test-utils";
import { beforeEach, describe, expect, it, vi } from "vitest";
import { key } from "../../../../src/store";
import routes from "../../../../src/router";
import Device from "../../../../src/views/Device.vue";

describe("Device", () => {
  const store = createStore({
    state: {},
    getters: {
      "devices/perPage": () => 10,
      "devices/page": () => 1,
      "devices/numberDevices": () => 1,
    },
    actions: {
      "devices/search": vi.fn(),
      "devices/fetch": vi.fn(),
      "snackbar/showSnackbarErrorAction": vi.fn(),
    },
  });
  let wrapper: VueWrapper<any>;

  beforeEach(() => {
    const vuetify = createVuetify();

    wrapper = mount(Device, {
      global: {
        plugins: [[store, key], vuetify, routes],
      },
    });
  });

  it("Is a Vue instance", () => {
    expect(wrapper).toBeTruthy();
  });

  it("Renders the component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Renders the template with default data", async () => {
    expect(wrapper.vm.filter).toBe("");
  });

  it("Must change the filter value when input change", async () => {
    expect(wrapper.vm.filter).toEqual("");
    await wrapper.find("input").setValue("ShellHub");
    expect(wrapper.vm.filter).toEqual("ShellHub");
  });

  it("Should render all the components in the screen", () => {
    expect(wrapper.find("h1").text()).toContain("Devices");
    expect(wrapper.find("input").exists()).toBe(true);
    expect(wrapper.find("[data-test='devices-list']").exists()).toBe(true);
  });
});
