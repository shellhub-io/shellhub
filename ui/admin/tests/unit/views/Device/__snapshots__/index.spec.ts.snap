// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Device > Renders the component 1`] = `
"<div class="d-flex flex-column justify-space-between align-center flex-sm-row">
  <h1>Devices</h1>
  <div class="v-spacer"></div>
  <div class="w-50">
    <div class="v-input v-input--horizontal v-input--density-comfortable v-theme--light v-locale--is-ltr v-text-field v-input--plain-underlined">
      <!---->
      <div class="v-input__control">
        <div class="v-field v-field--appended v-field--single-line v-field--variant-underlined v-theme--light v-locale--is-ltr">
          <div class="v-field__overlay"></div>
          <div class="v-field__loader">
            <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
              <!---->
              <div class="v-progress-linear__background bg-primary"></div>
              <div class="v-progress-linear__buffer bg-primary" style="width: 0%;"></div>
              <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                <div class="v-progress-linear__indeterminate">
                  <div class="v-progress-linear__indeterminate long bg-primary"></div>
                  <div class="v-progress-linear__indeterminate short bg-primary"></div>
                </div>
              </transition-stub>
              <!---->
            </div>
          </div>
          <!---->
          <div class="v-field__field" data-no-activator="">
            <!----><label class="v-label v-field-label" for="input-v-0">
              <!---->Search by hostname
            </label>
            <!----><input size="1" type="text" id="input-v-0" class="v-field__input" value="">
            <!---->
          </div>
          <!---->
          <div class="v-field__append-inner">
            <!----><i class="mdi-magnify mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i>
          </div>
          <div class="v-field__outline">
            <!---->
            <!---->
          </div>
        </div>
      </div>
      <!---->
      <!---->
    </div>
  </div>
  <div class="v-spacer"></div>
</div>
<div class="v-card v-theme--light v-card--density-default v-card--variant-elevated mt-2">
  <!---->
  <div class="v-card__loader">
    <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
      <!---->
      <div class="v-progress-linear__background"></div>
      <div class="v-progress-linear__buffer" style="width: 0%;"></div>
      <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
        <div class="v-progress-linear__indeterminate">
          <div class="v-progress-linear__indeterminate long"></div>
          <div class="v-progress-linear__indeterminate short"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div data-v-aca44803="">
    <div data-v-aca44803="" data-test="devices-list">
      <div class="v-table v-theme--light v-table--density-default bg-background border rounded text-center">
        <!---->
        <div class="v-table__wrapper">
          <table>
            <thead class="bg-v-theme-background">
              <tr>
                <th class="text-center"><span tabindex="0" class="cursor-pointer text-decoration-underline" aria-describedby="v-tooltip-v-2">Online <!----><!--teleport start--><!--teleport end--></span></th>
                <th class="text-center"><span tabindex="0" class="cursor-pointer text-decoration-underline" aria-describedby="v-tooltip-v-3">Hostname <!----><!--teleport start--><!--teleport end--></span></th>
                <th class="text-center"><span tabindex="0" class="cursor-pointer text-decoration-underline" aria-describedby="v-tooltip-v-4">Info <!----><!--teleport start--><!--teleport end--></span></th>
                <th class="text-center"><span tabindex="0" class="cursor-pointer text-decoration-underline" aria-describedby="v-tooltip-v-5">Namespace <!----><!--teleport start--><!--teleport end--></span></th>
                <th class="text-center"><span>Tags</span></th>
                <th class="text-center"><span tabindex="0" class="cursor-pointer text-decoration-underline" aria-describedby="v-tooltip-v-6">Last Seen <!----><!--teleport start--><!--teleport end--></span></th>
                <th class="text-center"><span tabindex="0" class="cursor-pointer text-decoration-underline" aria-describedby="v-tooltip-v-7">Status <!----><!--teleport start--><!--teleport end--></span></th>
                <th class="text-center"><span>Actions</span></th>
              </tr>
            </thead>
            <tbody class="pa-4 text-subtitle-2">
              <tr>
                <td colspan="8" class="pa-4 text-subtitle-2 text-center"> No data available </td>
              </tr>
            </tbody>
          </table>
        </div>
        <!---->
      </div>
      <!--v-if-->
      <!--v-if-->
      <div class="d-flex w-100 justify-end align-center"><span class="text-subtitle-2 mr-4">Items per page:</span>
        <div>
          <div class="v-input v-input--horizontal v-input--density-default v-theme--light v-locale--is-ltr v-input--dirty v-text-field v-input--plain-underlined v-combobox v-combobox--single mb-4">
            <!---->
            <div class="v-input__control">
              <div class="v-field v-field--active v-field--appended v-field--dirty v-field--no-label v-field--variant-underlined v-theme--light v-locale--is-ltr" role="combobox" aria-haspopup="menu" aria-expanded="false" aria-controls="v-menu-v-10">
                <div class="v-field__overlay"></div>
                <div class="v-field__loader">
                  <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                    <!---->
                    <div class="v-progress-linear__background"></div>
                    <div class="v-progress-linear__buffer" style="width: 0%;"></div>
                    <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                      <div class="v-progress-linear__indeterminate">
                        <div class="v-progress-linear__indeterminate long"></div>
                        <div class="v-progress-linear__indeterminate short"></div>
                      </div>
                    </transition-stub>
                    <!---->
                  </div>
                </div>
                <!---->
                <div class="v-field__field" data-no-activator="">
                  <!---->
                  <!---->
                  <!---->
                  <div class="v-field__input" data-no-activator="">
                    <!---->
                    <!---->
                    <div class="v-combobox__selection"><span class="v-combobox__selection-text">10<!----></span></div><input size="1" role="combobox" type="text" id="input-v-8" outlined="" value="10">
                  </div>
                  <!---->
                </div>
                <!---->
                <div class="v-field__append-inner">
                  <!----><i class="mdi-menu-down mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable v-combobox__menu-icon" role="button" aria-hidden="false" tabindex="-1" aria-label="Open" title="Open"></i>
                  <!---->
                </div>
                <div class="v-field__outline">
                  <!---->
                  <!---->
                </div>
              </div>
            </div>
            <!---->
            <!---->
          </div>
        </div>
        <div class="d-flex align-center"><button type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-plain" disabled=""><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
            <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-chevron-left mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
            <!---->
            <!---->
          </button><span class="text-subtitle-2">1 of 1</span><button type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-plain" disabled=""><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
            <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-chevron-right mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
            <!---->
            <!---->
          </button></div>
      </div>
    </div>
  </div>
  <!---->
  <!----><span class="v-card__underlay"></span>
</div>"
`;
